packages/nimble-components/src/select/index.ts

```diff
@@ -39,61 +89,776 @@ export class Select extends FoundationSelect implements ErrorPattern {
+            : Math.trunc(availableBottom);
         this.updateListboxMaxHeightCssVariable();
     }
 
-    // Workaround for https://github.com/microsoft/fast/issues/5773
     public override slottedOptionsChanged(
         prev: Element[],
         next: Element[]
     ): void {
         const value = this.value;
+        this._options.forEach(o => {
+            const notifier = Observable.getNotifier(o);
+            notifier.unsubscribe(this, 'value');
+        });
+
         super.slottedOptionsChanged(prev, next);
+
+        this._options.forEach(o => {
+            const notifier = Observable.getNotifier(o);
+            notifier.subscribe(this, 'value');
+        });
+        this.setProxyOptions();
+        this.updateValue();
+        // We need to force an update to the filteredOptions observable
+        // (by calling 'filterOptions()) so that the template correctly updates.
+        this.filterOptions();
         if (value) {
             this.value = value;
         }
+        this.committedSelectedOption = this.options[this.selectedIndex];
     }
 
-    private regionChanged(
-        _prev: AnchoredRegion | undefined,
-        _next: AnchoredRegion | undefined
+    public override clickHandler(e: MouseEvent): BooleanOrVoid {
+        // do nothing if the select is disabled
+        if (this.disabled) {
+            return;
+        }
+
+        if (this.open) {
+            const captured = (e.target as HTMLElement).closest<ListOption>(
+                'option,[role=option]'
+            );
+
+            if (!captured?.disabled) {
+                this.updateSelectedIndexFromFilteredSet();
+            }
+
+            if (captured?.disabled) {
+                return;
+            }
+        }
+
+        super.clickHandler(e);
+
+        this.open = this.collapsible && !this.open;
+
+        if (!this.open && this.indexWhenOpened !== this.selectedIndex) {
+            this.updateValue(true);
+        }
+    }
+
+    /**
+     * Updates the value when an option's value changes.
+     *
+     * @param source - the source object
+     * @param propertyName - the property to evaluate
+     *
+     * @internal
+     * @override
+     */
+    public override handleChange(source: unknown, propertyName: string): void {
+        super.handleChange(source, propertyName);
+        if (propertyName === 'value') {
+            this.updateValue();
+        }
+    }
+
+    /**
+     * Prevents focus when size is set and a scrollbar is clicked.
+     *
+     * @param e - the mouse event object
+     *
+     * @override
+     * @internal
+     */
+    public override mousedownHandler(e: MouseEvent): BooleanOrVoid {
+        if (e.offsetX >= 0 && e.offsetX <= this.listbox?.scrollWidth) {
+            return super.mousedownHandler(e);
+        }
+
+        return this.collapsible;
+    }
+
+    /**
+     * @ainternal
+     */
+    public regionLoadedHandler(): void {
+        this.focusAndScrollOptionIntoView();
+    }
```

Comment:
@rajsite @jattasNI @mollykreis, I've added this handler is response to noticing that in my attempts to integrate the filterable select into SLE, in the instance I was updating the text focus was not given to the filter input the first time you opened the dropdown (a fairly significant problem). By adding this handler, coupled with a style change I already had in place to use `display: none` instead of `visibility: hidden` to control the visibility of the dropdown, I was able to get the `Select` to behave as expected. Note that in our own Angular example app, I was _not_ able to replicate the same problem.

Unfortunately, I'm not aware of a good way to test this specific implementation, particularly with respect to solving the issue at hand. I propose we move forward with this implementation anyway, as it should be innocuous at worst.